### DISCUSSION TITLE
[REFACTOR]

### DISCUSSION THREAD

**Discussion by lukaszraczylo** [longhorn/longhorn#5525] [General]
## Is your improvement request related to a feature? Please describe

At the moment most of the images used within deploy directory use the head-master tag with image pull policy set to ifNotPresent.

This causes some issues for the "long term" users of the longhorn, most likely using outdated images versions. 

As an example in my case - despite of rocking flawlessly working latest rc1, the volumes were missing compressionMethod parameter ( from top of my head ) and this caused multiple failures. After some debugging session I've realised that my master-head was at least few months old.

A clear and concise description of what the problem is.

## Describe the solution you'd like

It would be great if either additional overlay for kustomize would be applied, overwriting the images versions to the current ones OR the imagePullPolicy would be set to Always.

The second option is a bit less desirable as it may increase the "fluff" traffic

**Comment by innobead** [comment]
@lukaszraczylo are you using the master head? which is under development not for production usage, like the compression method for backup is for 1.5 instead of 1.4. 

Suggest using the official release versions instead. (or any purpose to use the master head instead of official release)

**Comment by lukaszraczylo** [comment]
1. Installed longhorn with kubectl apply -f on the deploy directory with the 1.4-rc1 tag
2. Whatever was installed / introduced came from that installation and/or containers deployed during it

I think this confusion highlights the danger of the master-head and makes my point a bit stronger :)

**Comment by shuo-wu** [comment]
For any official images, we won't change them after pushing them to Docker HUB. Hence it's safe to use policy `IfNotPresent`.

When we deploy the dev version (`master-head`) or other dev versions, personally I will manually apply some changes including modifying the policy (both in the YAML file **and the default setting**) and frontend config... Typically it's not complicated : )
